FlowCrypt/Controllers/Threads/ThreadDetailsViewController+TapActions.swift

```diff
@@ -153,6 +159,28 @@ extension ThreadDetailsViewController {
         let attachments = quoteType == .forward
             ? input.processedMessage?.attachments ?? []
             : []
+        showSpinner()
+        let decryptedAttachments = try await attachments.asyncMap { (attachment: MessageAttachment) -> MessageAttachment in
+            var updatedAttachment = attachment
+            if updatedAttachment.data == nil {
+                updatedAttachment.data = try await messageHelper.download(
+                    attachment: updatedAttachment,
+                    messageId: input.rawMessage.identifier,
+                    progressHandler: { [weak self] progress in
+                        self?.handleFetchProgress(state: .download(progress))
+                    }
+                )
+            }
+            if updatedAttachment.isEncrypted {
+                handleFetchProgress(state: .decrypt)
+                return try await messageHelper.decrypt(
+                    attachment: updatedAttachment,
+                    userEmail: appContext.user.email
+                )
+            }
+            return updatedAttachment
```

Comment:
We have almost the same code in `getAttachment(at indexPath: IndexPath)`, may be it's possible to move it to separate method `fetchAndDecrypt(attachment: MessageAttachment)` to avoid duplication?

Then it'll be just 
```swift
let decryptedAttachments = try await attachments.asyncMap { fetchAndDecrypt(attachment: $0)
```